src/cogent3/core/alignment.py

```diff
@@ -2013,6 +2013,22 @@ def __init__(self, map, data, length=None):
         if hasattr(data, "name"):
             self.name = data.name
 
+    def annotate_matches_to(
+        self, pattern=None, annot_type=None, name=None, allow_multiple=False
+    ):
+        """Creates an annotation at the specified pattern in a sequence."""
+        from cogent3.core.annotation import Feature
+
+        pos = [m.span() for m in re.finditer(pattern, self._seq)]
+        if allow_multiple == False:
+            annot = self.add_annotation(Feature, annot_type, name, pos)
+        else:
+            for i in range(0, len(pos)):
```

Comment:
change this to be `annot = [self.add_annotation(
                    Feature, annot_type, f"{name}:{i}", [pos[i]]
                ) ...]`